### PR TITLE
Update gradle to version 8.0.1

### PR DESCRIPTION
**Pull request by c00ler**

### CODE REVIEW

**Review by guylabs — APPROVED** [review]

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 plugins {
     id("com.gradle.enterprise") version "3.12.3"
     id("com.gradle.common-custom-user-data-gradle-plugin") version "1.8.2"
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.4.0"
```
Comment (guylabs) [inline settings.gradle.kts:4]: Why is that required for the Gradle upgrade?

File: gradle.properties [hunk 2]
```diff
@@ -1,2 +1,3 @@
 version = 2.3-SNAPSHOT
 githubUser = jenkinsci
+org.gradle.jvmargs=-Xmx1g -XX:MaxMetaspaceSize=512m
```
Comment (guylabs) [inline gradle.properties:3]: Keep in mind that this is also applied to any CI agents where this project is running on. Does it mean that with Gradle 8 you get any kind of OOM errors?

**Review by welandaz — APPROVED** [review]

**Review by sorin-florea — APPROVED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (c00ler) [reply settings.gradle.kts:4]: I was also surprised, but auto-discovery was not working properly on jenkins: https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fgradle-plugin/detail/PR-243/6/pipeline/21

File: gradle.properties (hunk 2 above)
Comment (c00ler) [reply gradle.properties:3]: Yes, exactly, with the standard memory limits it was not working on jenkins.
